### ISSUE TITLE
CreateTempTableAsync with Postgres does not handle identity columns as expected

### ISSUE THREAD

**Issue by bt-88** [linq2db/linq2db#4333]
### Describe your issue

When creating a temporary table via DataConnection.CreateTempTableAsync<T>(), using `PostgreSQLDataProvider93`, the generated SQL does not take into account IDENTITY fields in my entity.

Therefore, when attempting to (bulk)copy records into that table it fails with `null value in column "id" of relation XXX` because it expects a value for the identity column.

See the unit test below that reproduces this issue.

```
Exception message: pgsql.PostgresException (0x80004005): 23502: null value in column "id" of relation "ledgers" violates not-null constraint
```

### Steps to reproduce

```c#
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using DotNet.Testcontainers.Builders;
using LinqToDB;
using LinqToDB.Data;
using LinqToDB.EntityFrameworkCore;
using Microsoft.EntityFrameworkCore;
using Shouldly;
using Testcontainers.PostgreSql;

namespace Fusion.Common.Tests;

public class Linq2DbPostgresTests : IAsyncLifetime
{
    private readonly PostgreSqlContainer _pgsqlContainer = new PostgreSqlBuilder()
        .WithImage("postgres:latest")
        //.WithDatabase("fusion")
        .WithPortBinding(5432, true)
        //.WithPassword("postgres")
        //.WithUsername("postgres")
        .WithWaitStrategy(Wait.ForUnixContainer().UntilPortIsAvailable(5432))
        .Build();
    
    internal class Foo
    {
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)] public int Id { get; private set; }
        [MaxLength(50)] public required string Name { get; init; }
    }
    
    internal class FooDbContext : DbContext
    {
        public FooDbContext(DbContextOptions options) : base(options) { }
        
        private DbSet<Foo> Foos { get; init; } = null!;
    }

    private FooDbContext CreateFooDbContext()
    {
        var options = new DbContextOptionsBuilder()
            .UseNpgsql(_pgsqlContainer.GetConnectionString())
            .UseSnakeCaseNamingConvention()
            .Options;

        return new FooDbContext(options);
    }
    
    public Task InitializeAsync() => _pgsqlContainer.StartAsync();
    public Task DisposeAsync() => _pgsqlContainer.StopAsync();

    [Fact]
    public async Task CreateTempTable_AndFillIt_ShouldNotFail()
    {
        await using var dbContext = CreateFooDbContext();
        var db = dbContext.CreateLinqToDBConnection();
        
        // Create temporary table
        var table = await db.CreateTempTableAsync<Foo>(tableOptions: TableOptions.IsTemporary);
        
        /* Generate SQL statement my NpgsqlProvider93:
         
         CREATE TEMPORARY TABLE foos
            (
	            id   integer                NOT NULL,
	            name character varying(50)  NOT NULL,

	            CONSTRAINT "PK_foos" PRIMARY KEY (id)
            )
            ON COMMIT PRESERVE ROWS
         */

        // Create data
        var data = new[]
        {
            new Foo { Name = "Bar" },
            new Foo { Name = "Baz" },
        };
        
        // Try to copy data to temp table
        // This will throw an Npgsql.PostgresException (0x80004005: 23502: null value in column "id" of relation "foos" violates not-null contraint
        var result = await table.BulkCopyAsync(data);
        
        result.RowsCopied.ShouldBe(2);
    }
}
```

### Environment details

`Linq To DB` version: 7.5.0
Database (with version): postgres:latest (16.1) on Docker
ADO.NET Provider (with version): PostgreSQLDataProvider93
Operating system: Windows 10.0.22621 Build 22621
.NET Version: 7

**Comment by to11mtm** [comment]
Can you please provide Linq To DB , Linq To DB EfCore , and EFCore versions?

That said, at least at first glance, this may indicate an issue with how LinqToDB EFCore is handling the Identity conversions.
I don't _think_ it will help, but have you tried declaring the identity column as a `long` instead?

**Comment by bt-88** [comment]
> Can you please provide Linq To DB , Linq To DB EfCore , and EFCore versions?
> 
> That said, at least at first glance, this may indicate an issue with how LinqToDB EFCore is handling the Identity conversions. I don't _think_ it will help, but have you tried declaring the identity column as a `long` instead?

Linq2Db `7.5.0`
Linq2Db.EFCore `7.5.0`
Microsoft.EntityFrameworkCore `7.0.11`
Npgsql.EntityFrameworkCore.PostgreSQL `7.0.11`

**Comment by avalara-stephen-hickey** [comment]
I am also experiencing this issue. I can also provide a minimal example, but the only real difference is I am using an `EntityTypeConfiguration` instead of annotations. I've tried both `.UseIdentityAlwaysColumn();` and `.ValueGeneratedOnAdd()`

> Can you please provide Linq To DB , Linq To DB EfCore , and EFCore versions?

linq2db `5.3.1`
linq2db.EntityFrameworkCore `8.0.0-rc.2`
Microsoft.EntityFrameworkCore `8.0.0`
Npgsql.EntityFrameworkCore.PostgreSQL `8.0.0`

> have you tried declaring the identity column as a long instead?
Yes, this occurs with `long` too.

**Comment by bt-88** [comment]
> I am also experiencing this issue. I can also provide a minimal example, but the only real difference is I am using an `EntityTypeConfiguration` instead of annotations and using `.CreateTableAsync` (instead of `CreateTempTableAsync`). I've tried both `.UseIdentityAlwaysColumn();` and `.ValueGeneratedOnAdd()`
> 
> > Can you please provide Linq To DB , Linq To DB EfCore , and EFCore versions?
> 
> linq2db `5.3.1` linq2db.EntityFrameworkCore `8.0.0-rc.2` Microsoft.EntityFrameworkCore `8.0.0` Npgsql.EntityFrameworkCore.PostgreSQL `8.0.0`
> 
> > have you tried declaring the identity column as a long instead?
> 
> Yes, this issue occurs with `long` too.

I have found a work around for my use case.

Instead of merging from a temp table (via either `CreateTableAsync` or `CreateTempTableAsync`) and filling that table with `IEnumerable<Foo>`, I now setup my merge directly from an `IEnumerable<Foo>` which does work.

So instead of doing `.Using(myTempTable)`, I do `.Using(myItems)`.

I should probably have a look at the logs what kinds of queries this produces on the db -- I presume that it creates a temp table for this -- but it works and performance is excellent.

**Comment by avalara-stephen-hickey** [comment]
My current work-around is to run raw SQL after the table is created that alters the column to add identity if it doesn't already exist. Pretty hacky but what else can you do?

**Comment by sdanyliv** [comment]
Looks like a bug. It is also hacky for us how to retrieve Indentity information from each provider.

**Comment by MaceWindu** [comment]
As workaround add linq2db `[Identity]` attribute

**Comment by bt-88** [comment]
@MaceWindu: Nope. Fails for the same reason on Postgres. Please see my test in the beginning of this topic.

![image](https://github.com/user-attachments/assets/262ea4c7-891a-4a22-a021-fd2f911e8f6d)
